src/Npgsql/ThrowHelper.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Reflection;
+
+namespace Npgsql
+{
+    static class ThrowHelper
+    {
+        internal static void ThrowNotSupportedException_NoPropertyGetter(Type type, MemberInfo property) =>
```

Comment:
If I understand correctly, you want to introduce something like [this pattern](https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/ThrowHelper.cs) into Npgsql, right?

If this is for performance, isn't the idea to have a non-inlinable method that does the actual throwing, to keep the size of the calling method down? Also, are we sure that this actually helps performance on newer JITs?

From a readability/codebase perspective, I prefer to just have exceptions thrown inline in the method - I think this makes everything clearer and easier to understand. If there's a performance aspect, I'd like to understand it and be sure it's still valid...

Reply:
See #2237.

Reply:
Thanks for reminding me, continuing the conversation in that issue.